modules/openapi-generator/src/main/resources/typescript/configuration.mustache

```diff
@@ -99,3 +99,66 @@ export function createConfiguration(conf: ConfigurationParameters = {}): Configu
     }
     return configuration;
 }
+
+/**
+ * Merge configuration options into a configuration.
+ */
+{{#useInversify}}
+export function mergeConfiguration(conf: Configuration, options?: Configuration): Configuration {
+    if (options) {
+        conf = options;
+    }
+    return conf;
+}
+{{/useInversify}}
+{{^useInversify}}
+export function mergeConfiguration(conf: Configuration, options?: ConfigurationOptions): Configuration {
+    let allMiddleware: Middleware[] = [];
+    if (options && options.middleware) {
+        const middlewareMergeStrategy = options.middlewareMergeStrategy || "replace" // default to replace behavior
+        // call-time middleware provided
+        const calltimeMiddleware: Middleware[] = options.middleware;
+
+        switch(middlewareMergeStrategy) {
+        case "append":
+            allMiddleware = conf.middleware.concat(calltimeMiddleware);
+            break;
+        case "prepend":
+            allMiddleware = calltimeMiddleware.concat(conf.middleware)
+            break;
+        case "replace":
+            allMiddleware = calltimeMiddleware
+            break;
+        default:
+            throw new Error(`unrecognized middleware merge strategy '${middlewareMergeStrategy}'`);
+        }
+    }
+    if (options) {
+        conf = {
+          baseServer: options.baseServer || conf.baseServer,
+          httpApi: options.httpApi || conf.httpApi,
+          authMethods: options.authMethods || conf.authMethods,
+          middleware: allMiddleware || conf.middleware
+        };
+    }
+    return conf;
+}
+{{/useInversify}}
+
+/**
+ * Convert Promise-based configuration options to Observable-based configuration options.
+ */
+export function wrapOptions(options?: PromiseConfigurationOptions): StandardConfigurationOptions | undefined {
+    if (options) {
+        return {
+            baseServer: options.baseServer,
+            httpApi: options.httpApi,
+            middleware: options.middleware?.map(
+              m => new PromiseMiddlewareWrapper(m)
+            ),
+            middlewareMergeStrategy: options.middlewareMergeStrategy,
+            authMethods: options.authMethods,
+        }
+    }
+    return;
```

Comment:
superfluous?

Reply:
Typescript requires an explicit return on all code paths.

Reply:
TIL, we had this enabled. Not personally a fan, but 🤷 . Thanks for explaining.